services/API-service/src/api/admin-area-dynamic-data/admin-area-dynamic-data.service.ts

```diff
@@ -130,47 +137,51 @@ export class AdminAreaDynamicDataService {
 
   private async deleteDynamicDuplicates(
     uploadExposure: UploadAdminAreaDynamicDataDto,
-  ): Promise<void> {
+  ): Promise<DeleteResult> {
+    const uploadCutoffMoment = this.helperService.getUploadCutoffMoment(
+      uploadExposure.disasterType,
+      uploadExposure.date,
+    );
+
     const deleteFilters = {
       indicator: uploadExposure.dynamicIndicator,
       countryCodeISO3: uploadExposure.countryCodeISO3,
       adminLevel: uploadExposure.adminLevel,
       disasterType: uploadExposure.disasterType,
-      timestamp: MoreThanOrEqual(
-        this.helperService.getUploadCutoffMoment(
-          uploadExposure.disasterType,
-          uploadExposure.date,
-        ),
-      ),
+      timestamp: MoreThanOrEqual(uploadCutoffMoment),
     };
     if (uploadExposure.eventName) {
       deleteFilters['eventName'] = uploadExposure.eventName;
     }
-    await this.adminAreaDynamicDataRepo.delete(deleteFilters);
+
+    return this.adminAreaDynamicDataRepo.delete(deleteFilters);
   }
 
   private async insertAlertPerLeadTime(
     uploadExposure: UploadAdminAreaDynamicDataDto,
-  ): Promise<void> {
+  ): Promise<AlertPerLeadTimeEntity[]> {
     const forecastTrigger = this.isForecastTrigger(
       uploadExposure.exposurePlaceCodes,
     );
 
     const forecastAlert = !!uploadExposure.eventName; // NOTE AB#32041: eventName being filled or not should no longer be needed to distinguish alert/warning from no alert.
 
-    const uploadAlertPerLeadTimeDto = new UploadAlertPerLeadTimeDto();
-    uploadAlertPerLeadTimeDto.countryCodeISO3 = uploadExposure.countryCodeISO3;
-    uploadAlertPerLeadTimeDto.disasterType = uploadExposure.disasterType;
-    uploadAlertPerLeadTimeDto.eventName = uploadExposure.eventName;
-    uploadAlertPerLeadTimeDto.alertsPerLeadTime = [
+    const uploadAlertsPerLeadTimeDto = new UploadAlertsPerLeadTimeDto();
```

Comment:
sounds good. Let's hope the rebasing will remain manageable with all these changes on code I'm also touching..